### ISSUE TITLE
1-org deployment issue

### ISSUE THREAD

**Issue by Atul7696** [terraform-google-modules/terraform-example-foundation#768]
### TL;DR

tfplan not gettng generated at step no 16 of 1-org documentation

### Expected behavior

tf plan should create

### Observed behavior

can you advise...all directory structure is as what is cloned....have tried with all service accounts organization-sa , terraform-org-sa and terraform-net-sa as well.. Any help will be highly appreciated

### Terraform Configuration

```hcl
Configurations are same as listed in #763. Need urgent advise on this
```

### Terraform Version

```sh
Terraform v0.13.7
```

### Additional information

Configurations are same as listed in #763. Need urgent advise on this.
gcp-org and gcp-policies are cloned at the same level as terraform-example-foundation.
have tried inside terraform-example-foundation.
0-bootstrap was successful
have tried with terraform-network-sa account as well as we are enabling hub and spoke model.
have used org-terrform account and with terraform-org-sa 

Please provide help in this

**Comment by Atul7696** [comment]
@dgulli and @organic_data @mvcap any thoughts on this. I am getting similar error and logs as mentioned in issue #619

**Comment by Atul7696** [comment]
Below are the logs from cloud build

starting build "ffd1d4db-7c5c-4b7d-b3e7-0aae781b91e1"

FETCHSOURCE
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
Initialized empty Git repository in /workspace/.git/
From https://source.developers.google.com/p/prj-b-cicd-1b28/r/gcp-org
 * branch            0c769ddc3c42c5ead94379b799ec045360197035 -> FETCH_HEAD
HEAD is now at 0c769dd aa
BUILD
Starting Step #0 - "setup"
Step #0 - "setup": Pulling image: us-east4-docker.pkg.dev/prj-b-cicd-1b28/prj-tf-runners/terraform
Step #0 - "setup": Using default tag: latest
Step #0 - "setup": latest: Pulling from prj-b-cicd-1b28/prj-tf-runners/terraform
Step #0 - "setup": c1ad9731b2c7: Pulling fs layer
Step #0 - "setup": aac672a3304c: Pulling fs layer
Step #0 - "setup": 695472ca2b70: Pulling fs layer
Step #0 - "setup": 28cb0f77f9cc: Pulling fs layer
Step #0 - "setup": 7d2962e457f4: Pulling fs layer
Step #0 - "setup": 0b20c6d36136: Pulling fs layer
Step #0 - "setup": a69e48b9276e: Pulling fs layer
Step #0 - "setup": 28cb0f77f9cc: Waiting
Step #0 - "setup": 7d2962e457f4: Waiting
Step #0 - "setup": 0b20c6d36136: Waiting
Step #0 - "setup": a69e48b9276e: Waiting
Step #0 - "setup": aac672a3304c: Download complete
Step #0 - "setup": c1ad9731b2c7: Verifying Checksum
Step #0 - "setup": c1ad9731b2c7: Download complete
Step #0 - "setup": 7d2962e457f4: Verifying Checksum
Step #0 - "setup": 7d2962e457f4: Download complete
Step #0 - "setup": 0b20c6d36136: Verifying Checksum
Step #0 - "setup": 0b20c6d36136: Download complete
Step #0 - "setup": a69e48b9276e: Verifying Checksum
Step #0 - "setup": a69e48b9276e: Download complete
Step #0 - "setup": c1ad9731b2c7: Pull complete
Step #0 - "setup": aac672a3304c: Pull complete
Step #0 - "setup": 28cb0f77f9cc: Verifying Checksum
Step #0 - "setup": 28cb0f77f9cc: Download complete
Step #0 - "setup": 695472ca2b70: Verifying Checksum
Step #0 - "setup": 695472ca2b70: Download complete
Step #0 - "setup": 695472ca2b70: Pull complete
Step #0 - "setup": 28cb0f77f9cc: Pull complete
Step #0 - "setup": 7d2962e457f4: Pull complete
Step #0 - "setup": 0b20c6d36136: Pull complete
Step #0 - "setup": a69e48b9276e: Pull complete
Step #0 - "setup": Digest: sha256:af457744b3db2fe2674970e0b0b14d481ebfe62028e683c4e5430b514198e41d
Step #0 - "setup": Status: Downloaded newer image for us-east4-docker.pkg.dev/prj-b-cicd-1b28/prj-tf-runners/terraform:latest
Step #0 - "setup": us-east4-docker.pkg.dev/prj-b-cicd-1b28/prj-tf-runners/terraform:latest
Step #0 - "setup": Setting up gcloud for impersonation
Step #0 - "setup": Updated property [auth/impersonate_service_account].
Step #0 - "setup": Adding bucket information to backends
Finished Step #0 - "setup"
Starting Step #1 - "tf init"
Step #1 - "tf init": Already have image (with digest): us-east4-docker.pkg.dev/prj-b-cicd-1b28/prj-tf-runners/terraform
Step #1 - "tf init": shared doesn't match plan; skipping
Step #1 - "tf init": refs doesn't match plan; skipping
Step #1 - "tf init": objects doesn't match plan; skipping
Step #1 - "tf init": logs doesn't match plan; skipping
Step #1 - "tf init": info doesn't match plan; skipping
Step #1 - "tf init": hooks doesn't match plan; skipping
Step #1 - "tf init": branches doesn't match plan; skipping
Finished Step #1 - "tf init"
Starting Step #2 - "tf plan"
Step #2 - "tf plan": Already have image (with digest): us-east4-docker.pkg.dev/prj-b-cicd-1b28/prj-tf-runners/terraform
Step #2 - "tf plan": shared doesn't match plan; skipping
Step #2 - "tf plan": refs doesn't match plan; skipping
Step #2 - "tf plan": objects doesn't match plan; skipping
Step #2 - "tf plan": logs doesn't match plan; skipping
Step #2 - "tf plan": info doesn't match plan; skipping
Step #2 - "tf plan": hooks doesn't match plan; skipping
Step #2 - "tf plan": branches doesn't match plan; skipping
Finished Step #2 - "tf plan"
Starting Step #3 - "tf validate"
Step #3 - "tf validate": Already have image (with digest): us-east4-docker.pkg.dev/prj-b-cicd-1b28/prj-tf-runners/terraform
Step #3 - "tf validate": shared doesn't match plan; skipping
Step #3 - "tf validate": refs doesn't match plan; skipping
Step #3 - "tf validate": objects doesn't match plan; skipping
Step #3 - "tf validate": logs doesn't match plan; skipping
Step #3 - "tf validate": info doesn't match plan; skipping
Step #3 - "tf validate": hooks doesn't match plan; skipping
Step #3 - "tf validate": branches doesn't match plan; skipping
Finished Step #3 - "tf validate"
Starting Step #4 - "tf apply"
Step #4 - "tf apply": Already have image (with digest): us-east4-docker.pkg.dev/prj-b-cicd-1b28/prj-tf-runners/terraform
Step #4 - "tf apply": shared doesn't match plan; skipping
Step #4 - "tf apply": refs doesn't match plan; skipping
Step #4 - "tf apply": objects doesn't match plan; skipping
Step #4 - "tf apply": logs doesn't match plan; skipping
Step #4 - "tf apply": info doesn't match plan; skipping
Step #4 - "tf apply": hooks doesn't match plan; skipping
Step #4 - "tf apply": branches doesn't match plan; skipping
Finished Step #4 - "tf apply"
PUSH
Artifacts will be uploaded to gs://prj-cloudbuild-artifacts-7446 using gsutil cp
cloudbuild-tf-apply.yaml: Uploading path....
Copying file://cloudbuild-tf-apply.yaml [Content-Type=application/octet-stream]...
/ [0/1 files][    0.0 B/  2.2 KiB]   0% Done                                    
/ [1/1 files][  2.2 KiB/  2.2 KiB] 100% Done                                    
Operation completed over 1 objects/2.2 KiB.                                      
cloudbuild-tf-apply.yaml: 1 matching files uploaded
tmp_plan/*.tfplan: Uploading path....
CommandException: No URLs matched: tmp_plan/*.tfplan
CommandException: 1 file/object could not be transferred.
ERROR
ERROR: could not upload tmp_plan/*.tfplan to gs://prj-cloudbuild-artifacts-7446/terraform/cloudbuild/apply/ffd1d4db-7c5c-4b7d-b3e7-0aae781b91e1/; err = gsutil exited with non-zero status: 1

**Comment by Atul7696** [comment]
./tf-wrapper.sh plan production                                                                                 
*************** TERRAFORM PLAN *******************
      At environment: envs/shared 
**************************************************
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

module.org_domain_restricted_sharing.data.google_organization.orgs["ourinnovationlabs.com"]: Refreshing state...

------------------------------------------------------------------------

Warning: Value for undeclared variable

The root module does not declare a variable named
"networks_step_terraform_service_account" but a value was found in file
"terraform.tfvars". To use this value, add a "variable" block to the
configuration.

Using a variables file to set an undeclared variable is deprecated and will
become an error in a future release. If you wish to provide certain "global"
settings to all configurations in your organization, use TF_VAR_...
environment variables to set these instead.


Error: Attribute must be a whole number, got 2.592e+09

  on .terraform/modules/bigquery_destination/modules/bigquery/main.tf line 47, in resource "google_bigquery_dataset" "dataset":
  47:   default_table_expiration_ms = var.expiration_days == null ? null : var.expiration_days * 864 * pow(10, 5)

**Comment by Atul7696** [comment]
Commented the line 47 .terraform/modules/bigquery_destination/modules/bigquery/main.tf line 47 and the plan was saved successfully locally in tmp_plan

Seems issue with the terraform module. this needs to be corrected

**Comment by Atul7696** [comment]
closing it as redeployed the Org structure after commented below lines in tf-wrapper.sh

 tf_validate "$env_path" "$env" "$policysource" "$component"
 gcloud beta terraform vet "${tf_env}.json" --policy-library="${policy_file_path}" --project="${project_id}" || exit 33
